src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java

```diff
@@ -197,4 +199,34 @@ public void setMaxChunkSizeToHsm(int maxChunkSizeToHsm) {
     public void setStopSending() {
         this.isStopped = true;
     }
+
+    @Override
+    public PowHSMBlockchainParameters getBlockchainParameters() throws HSMClientException {
+        final String BLOCKCHAIN_PARAMETERS_COMMAND = "blockchainParameters";
+        final String PARAMETERS_FIELD = "parameters";
+        final String CHECKPOINT_FIELD = "checkpoint";
+        final String MINIMUM_DIFFICULTY_FIELD = "minimum_difficulty";
+        final String NETWORK_FIELD = "network";
+
+        if (this.version < 3) {
+            throw new HSMUnsupportedTypeException("method call not allowed for version {}." + this.version);
+        }
+
+        ObjectNode command = this.hsmClientProtocol.buildCommand(BLOCKCHAIN_PARAMETERS_COMMAND, this.version);
+        JsonNode response = this.hsmClientProtocol.send(command);
+
+        this.hsmClientProtocol.validatePresenceOf(response, PARAMETERS_FIELD);
+        JsonNode parameters = response.get(PARAMETERS_FIELD);
+
+        this.hsmClientProtocol.validatePresenceOf(parameters, CHECKPOINT_FIELD);
+        this.hsmClientProtocol.validatePresenceOf(parameters, MINIMUM_DIFFICULTY_FIELD);
+        this.hsmClientProtocol.validatePresenceOf(parameters, NETWORK_FIELD);
+
+        String checkpoint = parameters.get(CHECKPOINT_FIELD).asText();
+        int minimumDifficulty = parameters.get(MINIMUM_DIFFICULTY_FIELD).asInt();
+        String network = parameters.get(NETWORK_FIELD).asText();
+
+        logger.info("[getBlockchainParameters] Checkpoint: {}, Minimum Difficulty: {}, Network: {}", checkpoint, minimumDifficulty, network);
+        return new PowHSMBlockchainParameters(checkpoint, minimumDifficulty, network);
```

Comment:
Does the checkpoint come with or without the `0x` prefix? Make sure it won't cause a problem when PowHSMBlockchainParameters constructor tries to convert it to Keccak

Reply:
It Comes Without The `0x` And Converts Properly To Keccak